fsspec/caching.py

```diff
@@ -49,7 +49,7 @@ class MMapCache(BaseCache):
 
     def __init__(self, blocksize, fetcher, size, location=None, blocks=None):
         super().__init__(blocksize, fetcher, size)
-        self.blocks = set() if blocks is None else blocks
+        self.blocks = set() if blocks is None else set(blocks)
```

Comment:
This fixes #333 on my local tests, but I did not have time to write a proper unit test for it.

The test needs to:
   - Open a block cache
   - Partially read the remote content.
   - Close the block cache
   - Reopen the block cache.
   - Ensure no exceptions are raised.

Not sure how to write that test yet though :(